shiny/ui/_card.py

```diff
@@ -130,6 +131,9 @@ def _card_impl(
     attrs, children = consolidate_attrs(*args, class_=class_, **kwargs)
     children = _wrap_children_in_card(*children, wrapper=wrapper)
 
+    if full_screen and "id" not in attrs:
+        attrs["id"] = f"bslib_card_{random.randint(1000, 10000)}"
```

Comment:
I think we want to use `private_random_int()` here, so that it wouldn't be influenced if the app code sets a random seed. I'd also suggest using a larger range of numbers -- if there are 5 items with `full_screen` on a page, then there is an approximately 1 in a 1000 chance of a name collision, which means it will happen with some regularity.

(Also, in the future we will probably want to make this repeatable/deterministic, but we can save that for another time.)

Reply:
I added `private_random_id()` to `_utils`, built around `rand_hex()` that makes ids using the private random seed that look like this:

```
>>> shiny._utils.private_random_id("bslib_card")
'bslib_card_b15a58'
```

which by default has a range of 2<sup>24</sup> possibilities (and is configurable via a `bytes` argument). I replaced the previous uses of `random.randint()` in the other places where it was already being used.

Reply:
Great! I like `private_random_id()`.

I was curious, so I wrote function to calculate the probability of a collision:

```R
# Calculate the probability of a collision if `n` items are randomly drawn (with
# replacement) from a set with `total` number of items.
collision_prob <- function(n, total) {
  prob_no_collision <- 1
  # Do this in a loop instead of prod(numerators)/prod(denominators), because
  # that method is prone to result in a loss of precision due to rounding.
  for (i in seq(0, n-1)) {
    prob_no_collision <- prob_no_collision * (total - i) / total
  }
  1 - prob_no_collision
}

collision_prob(5, 9000)
#> [1] 0.001110679

collision_prob(5, 2^24)
#> [1] 5.960463e-07

collision_prob(12, 2^24)
#> [1] 3.9339e-06
```

Even if we go up to 12 items, the probability of a collision is about 4 in a million, which I think is good enough for this use case.